drivers/adc-dac/ad5592r/ad5592r.c

```diff
@@ -131,6 +132,51 @@ int32_t ad5592r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
 	return 0;
 }
 
+/**
+ * Read Multiple ADC Channels.
+ *
+ * @param dev - The device structure.
+ * @param chans - The ADC channels to be readback
+ * @param values - ADC value array
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_multi_read_adc(struct ad5592r_dev *dev, uint16_t chans,
+			       uint16_t *values)
+{
+	int32_t ret;
+	uint8_t samples;
+	uint8_t i;
+
+	if (!dev)
+		return FAILURE;
```

Comment:
If you decided to use `error` defines, I would use them through the entire function.

Reply:
This is how errors were handled in other functions. I can change them, but that would require changing functions that I haven't changed too to keep things consistent. You're suggesting return SUCCESS rather than 0 or  alternatively changing FAILURE to -1?

Reply:
This driver is older, that's why the error codes are not used. These were introduced later on this repository. You can just return -1 here so you don't have to change the entire driver.

Same for `ad5593r_multi_read_adc`

Reply:
Return FAILURE is used elsewhere in the driver as well, in areas that I haven't altered. Should those be reverted too?

Reply:
I had a look on the initial commit of the driver, and indeed a combination of defines raw values was used from the start.

Leave it as it is if the driver was implemented in this manner.

Reply:
Perfect, will leave as is.